launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -296,29 +296,46 @@ QString FirstLaunchView::getHeroesInstallDir()
 void FirstLaunchView::extractGogData()
 {
 #ifdef ENABLE_INNOEXTRACT
-	QString filterExe = tr("GOG executable") + " (*.exe)";
-	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+
+	auto fileSelection = [this](QString type, QString filter, QString startPath = {}) {
+		QString titleSel = tr("Select a ") + filter + tr(" file...");
+		QString titleErr = tr("You have to select a ") + filter + tr(" file!");
```

Comment:
never do this for translatable strings, always use placeholders:
```suggestion
		QString titleSel = tr("Select %1 file...", "param is file extension").arg(filter);
		QString titleErr = tr("You have to select %1 file!", "param is file extension").arg(filter);
```

Reply:
Thanks! Understood.